operator/src/main/java/oracle/kubernetes/operator/DomainProcessorImpl.java

```diff
@@ -869,29 +866,9 @@ private boolean shouldReportAbortedEvent() {
       return Optional.ofNullable(eventData).map(EventData::getItem).orElse(null) == DOMAIN_PROCESSING_ABORTED;
     }
 
-    private void resetIntrospectorJobFailureCount() {
```

Comment:
Main already has functionality for resetting the failure count. Why discard it in favor of the approach used in 3.3.3?

Reply:
This method is resetting the failure count in the domain status. I believe this approach is same in latest 3.3.3 release. Previously we had 2 failure/retry counts (1) in the DomainStatus and (2) in-memory count in DomainPresenceInfo. After the fix for OWLS-90180, we can no longer rely on the in-memory state of the operator as the introspector job might have been created before the operator started. I have removed the in-memory retry count in DomainPresenceInfo and made changes to use the failure count in domain status instead.

Reply:
I hear the reasoning; I'm just surprised by the size of the change needed to do that.